syspage.h

```diff
@@ -19,112 +19,130 @@
+	union {
+		struct {
+			addr_t start;
+			addr_t end;
+		} blob;
 
-extern void syspage_setAddress(addr_t addr);
+		struct {
+			syspage_seg_t *segs;
+			addr_t entryPoint;
+		} progsegs;
+	};
 
+	char *argv;
 
-extern addr_t syspage_getAddress(void);
+	size_t imapSz;
+	u8 *imaps;
 
+	size_t dmapSz;
+	u8 *dmaps;
+} syspage_program_t;
 
-/* General functions */
 
-extern int syspage_save(void);
+typedef struct _syspage_map_t {
+	struct _syspage_map_t *next, *prev;
 
+	hal_memEntry_t *entries;
 
-extern void syspage_showAddr(void);
+	addr_t start;
+	addr_t end;
 
+	u32 attr;
+	u8 id;
 
-extern void syspage_showMaps(void);
+	char *name;
+} syspage_map_t;
 
 
-extern void syspage_showApps(void);
+typedef struct {
+	size_t size;
 
+	struct {
+		unsigned int minor;
+		unsigned int major;
+	} console;
 
-extern void syspage_showKernel(void);
+	syspage_map_t *maps;
+	syspage_program_t *progs;
+} syspage_t;
 
+#pragma pack(pop)
 
-/* Validation */
 
-extern int syspage_validateAddrMap(unsigned int ftype, addr_t addr, u8 id, unsigned int attr);
+/* General functions */
+extern void syspage_init(void);
 
 
-extern int syspage_validateKernel(addr_t *addr);
+extern void *syspage_alloc(size_t size);
 
 
 /* Map's functions */
-
-extern int syspage_addmap(const char *name, addr_t start, addr_t end, const char *attr);
+extern int syspage_addMap(const char *name, addr_t start, addr_t end, const char *attr);
 
 
-extern int syspage_getMapTop(const char *map, addr_t *addr);
+extern hal_memEntry_t *syspage_allocEntry(const char *mapName, addr_t start, size_t size, unsigned int align);
 
 
-extern int syspage_setMapTop(const char *map, addr_t addr);
+extern int syspage_getMapAttr(const char *name, unsigned int *attr);
 
 
-extern int syspage_alignMapTop(const char *map);
+extern int syspage_getMapID(const char *name, u8 *id);
 
 
-extern int syspage_getFreeSize(const char *map, size_t *sz);
-
-
-extern int syspage_write2Map(const char *map, const void *buff, size_t len);
-
-
-extern void syspage_addEntries(addr_t start, size_t sz);
-
-
-extern int syspage_getMapAttr(const char *map, unsigned int *attr);
-
-
-extern const char *syspage_getMapName(u32 id);
+extern const char *syspage_getMapName(u8 id);
 
 
 /* Program's functions */
-
-extern int syspage_addProg(addr_t start, addr_t end, const char *imap, const char *dmap, const char *name, u32 flags);
+extern syspage_program_t *syspage_allocProg(const char *argv, unsigned int type, u32 flags);
 
 
-/* Setting kernel's data */
+extern syspage_seg_t *syspage_allocSeg(syspage_program_t *prog);
 
-extern void syspage_setKernelEntry(addr_t addr);
 
+/* Console */
+extern void syspage_setConsole(unsigned int major, unsigned int minor);
 
-extern void syspage_setKernelText(addr_t addr, size_t size);
 
+/* Validation */
+extern int syspage_validateKernel(addr_t *addr);
 
-extern void syspage_setKernelBss(addr_t addr, size_t size);
 
+/* Information functions */
+extern void syspage_showMaps(void);
 
```

Comment:
I suggest to change the name of this function to `syspage_mapShow()`. The other suggested change is the semantics and add the `const char *name` argument. If it is defined (not `NULL`) all maps are presented. Otherwise only map selected by `name` argument is presented.